### PR TITLE
[WIP] Changes required to enable crossgen on the tests.

### PR DESCRIPTION
**Pull request by bryanar**
All issues removed. Pending completion of conversations below before I merge.

- Added crossgen and priority test building to the buildtest.cmd file (and updated the documentation as well for these things). Building crossgen tests is as easy as
> buildtest.cmd crossgen

Building different priority levels is as easy as
> buildtest.cmd priority 1

and doing a clean, priority 1 crossgen is as easy as
> buildtest.cmd clean priority 1 crossgen

### CODE REVIEW

File: tests/src/CLRTest.Execute.Batch.targets [hunk 1]
```diff
@@ -18,6 +18,33 @@ WARNING:   When setting properties based on their current state (for example:
...
+  <Target
+    Name="GetCrossgenBatchScript"
+    Returns="$(CrossgenBatchScript)">
+    
+    <PropertyGroup>
+
+<!-- CrossGen will create output if it needs to crossgen. Otherwise there will be silence. -->
+      <CrossgenBatchScript Condition="'$(CLRTestKind)' == 'BuildOnly' 
+                           OR '$(CLRTestKind)' == 'BuildAndRun' 
+                           OR '$(CLRTestKind)' == 'SharedLibrary'">
```
Comment (gkhanna79) [inline tests/src/CLRTest.Execute.Batch.targets]: What is "SharedLibrary" ?

**Comment by gkhanna79** [comment]
@mmitche - This appears to have the same infra issue as my PR. Do we understand this?

**Comment by gkhanna79** [comment]
At high-level, this looks good to me. Are you going to update the test wiki on how to run the tests with crossgen? Here is the link to the wiki:

https://github.com/dotnet/coreclr/blob/master/Documentation/building/windows-test-instructions.md

BTW, the wiki does not have P0/P1 test run information - you should add it as well, when authoring documentation for crossgen support.

@JohnChen0 Can you please take a look too?

**Comment by mmitche** [comment]
@gkhanna79 We don't know exactly what the issue is, but we know the fix is to remove the use of CodeTaskFactory (which may be necessary for the move to MSBuild CoreCLR anyway).  I had a hunch that a full TEMP dir makes the problem more persistent, but the TEMP dir on this machine was recently cleaned, so that may not be the case.

File: tests/src/dir.props [hunk 2]
```diff
@@ -11,6 +11,7 @@
     <NoWarn>78,162,164,168,169,219,251,252,414,429,642,649,652,675,1691,1717,1718,3001,3002,3003,3005,3008</NoWarn>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <CLRTestKind>BuildAndRun</CLRTestKind>
+    <SkipSigning Condition="'$(CrossGen)' == 'true'">true</SkipSigning>
```
Comment (JohnChen0) [inline tests/src/dir.props]: Why do we sign test apps when not doing CrossGen?

File: tests/src/CLRTest.Execute.Batch.targets [hunk 3]
```diff
@@ -18,6 +18,33 @@ WARNING:   When setting properties based on their current state (for example:
 -->
 <Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
+  <ItemGroup>
+    <CLRTestEnvironmentVariable Condition="'$(CrossGen)' == 'true'" Include="set complus_zaprequire=1" />
+    <CLRTestEnvironmentVariable Condition="'$(CrossGen)' == 'true'" Include="set complus_zaprequireexcludelist=corerun" />
```
Comment (JohnChen0) [inline tests/src/CLRTest.Execute.Batch.targets]: I'm surprised that it's necessary to add corerun to zaprequireexcludelist, since it's an unmanaged app.

**Comment by bryanar** [comment]
@gkhanna79 AHhhhhh you beat me to it. I want to make documenting part of my workflow on a regular basis. I will do this. For the record I *DID* create an additional document to describe the CLRTestPriority as well as the CLRTestKind and whatever, but it's sparse and not easy to find (obviously). I will try to plumb that through before I merge this. After lunch...

File: tests/src/dir.props (hunk 2 above)
Comment (bryanar) [inline tests/src/dir.props]: @weshaggard Wes do you know why we are signing?

Comment (weshaggard) [inline tests/src/dir.props]: I'm not aware of any particular reason we would be signing these test projects.

**Comment by bryanar** [comment]
@gkhanna79 

Added crossgen and priority test building to the buildtest.cmd file (and updated the documentation as well for these things). Building crossgen tests is as easy as > buildtest.cmd crossgen
Building different priority levels is as easy as

buildtest.cmd priority 1
and doing a clean, priority 1 crossgen is as easy as

buildtest.cmd clean priority 1 crossgen

**Pull request merged**